torch/utils/data/standard_pipes.ipynb

```diff
@@ -0,0 +1,1046 @@
+    "            yield i"
+   ]
+  },
+  {
+   "source": [
+    "## Batch\n",
+    "\n",
+    "Function: `batch`\n",
+    "\n",
+    "Description: \n",
+    "\n",
+    "Alternatives:\n",
+    "\n",
+    "Arguments:\n",
+    "  - `batch_size: int` desired batch size\n",
+    "  - `unbatch_level:int = 0` if specified calls `unbatch(unbatch_level=unbatch_level)` on source datapipe before batching (see `unbatch`)\n",
+    "  - `drop_last: bool = False`\n",
+    "\n",
+    "Example:\n",
+    "\n",
+    "Classic batching produce partial batches by default\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[0, 1, 2]\n[3, 4, 5]\n[6, 7, 8]\n[9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "To drop incomplete batches add `drop_last` argument"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[0, 1, 2]\n[3, 4, 5]\n[6, 7, 8]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3, drop_last = True)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "Sequential calling of `batch` produce nested batches"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[[0, 1, 2], [3, 4, 5]]\n[[6, 7, 8], [9, 10, 11]]\n[[12, 13, 14], [15, 16, 17]]\n[[18, 19, 20], [21, 22, 23]]\n[[24, 25, 26], [27, 28, 29]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(30).batch(3).batch(2)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "It is possible to unbatch source data before applying the new batching rule using `unbatch_level` argument"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]\n[10, 11, 12, 13, 14, 15, 16, 17, 18, 19]\n[20, 21, 22, 23, 24, 25, 26, 27, 28, 29]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(30).batch(3).batch(2).batch(10, unbatch_level=-1)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "## Unbatch\n",
+    "\n",
+    "Function: `unbatch`\n",
+    "\n",
+    "Description: \n",
+    "\n",
+    "Alternatives:\n",
+    "\n",
+    "Arguments:\n",
+    "    `unbatch_level:int = 1`\n",
+    " \n",
```

Comment:
Want to confirm with the behavior for `unbacth_level`
If the prior datapipe yields the following data (I am not sure if this should be expected, but it deserves to discuss if any user uses it)
```py
>>> for d in dp:
...     print(d)
[[1, 2], 3]
[[4, 5], 6]
[[7, 8], 9]
```
- `unbatch_level` is 1
```py
[1, 2]
3
[4, 5]
6
[7, 8]
9
```
- `unbatch_level` is -1 or 2
```py
1
2
3
4
...
```